webapp/src/pages/api/v1/tokens.ts

```diff
@@ -0,0 +1,43 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { faker } from '@faker-js/faker';
+import { TokenDto } from 'dogma/features/token/TokenDto';
+
+const newToken = (id: number) => {
+  const token: TokenDto = {
+    appId: `${faker.animal.snake().replaceAll(' ', '-').toLowerCase()}-${id}`,
+    secret: faker.datatype.uuid(),
+    admin: faker.datatype.boolean(),
+    creation: { user: faker.internet.email(), timestamp: faker.datatype.datetime().toISOString() },
+  };
+  if (id & 1) {
+    token.deactivation = { user: faker.internet.email(), timestamp: faker.datatype.datetime().toISOString() };
+  }
+  return token;
+};
+
+const tokenList: TokenDto[] = [];
+
+const makeData = (len: number) => {
+  for (let i = len; i > 0; i--) {
+    tokenList.push(newToken(i));
+  }
+};
+makeData(500);
```

Comment:
When 500 tokens are rendered, the response time for scroll, filter and so on looks slow.
🤔
Is it possible to apply an infinity-style scroll.

Reply:
Added the client side pagination. The only thing is resetting the filter needs to be done manually (using a button).

<img width="1224" alt="Screen Shot 2023-02-01 at 14 05 27" src="https://user-images.githubusercontent.com/5079588/215974499-a4d8767a-263e-4af5-96ab-12ad0ca46991.png">

Reply:
Wow. You make it so fast and the page is rendered fast as well.

I was wondering if it is possible to reset the filter when all search keywords are removed.
Additionally, I found out that the pagination moved back to the first page when I click next page after filtering tokens.

Reply:
In order for the pagination to not move back to the first page, we need the following line `value && column.setFilterValue(value)}`. 
Unfortunately, because of this work around we have to reset the filter by clicking the reset button.

Reply:
Found another way that doesn't require a reset button. 😅 Removed the button.

Reply:
Nice! 😀